common/ssh_max_arg.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python3
+# SPDX-FileCopyrightText: © 2015-2022 Germar Reitze
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# This file is part of the program "Back In time" which is released under GNU
+# General Public License v2 (GPLv2). See file/folder LICENSE or go to
+# <https://spdx.org/licenses/GPL-2.0-or-later.html>.
+"""This module determines the maximum possible length of an SSH command.
+
+It can also can run as a stand alone script. The solution is based on
+https://www.theeggeadventure.com/wikimedia/index.php/Ssh_argument_length
+"""
+import random
+import string
+import subprocess
+import socket
+import argparse
+from config import Config
+
+# Must be divisible by 8
+_INITIAL_SSH_CMD_SIZE = 1048320
+_ERR_CODE_E2BIG = 7
+
+
+def probe_max_ssh_command_size(
+        cfg: Config,
+        ssh_command_size: int = _INITIAL_SSH_CMD_SIZE,
+        size_offset: int = _INITIAL_SSH_CMD_SIZE) -> int:
+    """Determine the maximum length of SSH commands for the current config
+
+    Try a SSH command with length ``ssh_command_size``. The command is
+    decreased by ``size_offset`` if it was too long or increased if it worked.
+    The function calls itself recursively until it finds the maximum
+    possible length. The offset ``size_offset`` is bisect in each try.
+
+    Args:
+        config Back In Time config instance including the details about the
```

Comment:
Missing a colon (`:`) after `config`.
```suggestion
        config: Back In Time config instance including the details about the
```